.github/workflows/cmake.yml

```diff
@@ -10,10 +10,22 @@ env:
 
 jobs:
   build_and_test:
-    runs-on: ubuntu-20.04
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-20.04
+            allow-test-failure: false
+            # There sure is a lot of logic here, is there a better way?
+            should-notify-irc: ${{ github.repository == 'SerenityOS/serenity' && (github.event_name == 'pull_request' || (github.event_name == 'push' && github.ref == 'refs/heads/master')) }}
```

Comment:
TODO for the far future: Make IRC-notification a separate job that unconditionally runs after the build jobs.
Suggestion: None, I just wanted to make a note :)

Reply:
Yeah, unfortunately, in order to use the "nice" `on` syntax, it would have to be a separate workflow. which... ew.

Reply:
*Shocked pikachu face*

Oh well.

Reply:
Added a todo in the code for this